In [128]:
import pandas as pd
import numpy as np
import random
import os
from openpyxl import Workbook
from openpyxl import load_workbook
from scipy.stats import pearsonr
from sklearn.metrics import mean_squared_error

In [88]:
real_data_raw = pd.read_excel(r'../../Data/CompInfo/企业评分.xlsx')
output_path = r'../../Evaluation/mission1_predict.xlsx'

In [89]:
real_data_raw.head(n=2)

企业编号      企业总评分
0  1001  75.374276
1  1002  79.830122

In [90]:
real_data = real_data_raw.copy()

In [91]:
rate = real_data['企业总评分']
rate = rate.apply(np.round)
real_data['企业总评分'] = rate.astype(int)

In [92]:
real_data.loc[(real_data['企业编号'] == 1001), '企业总评分']-2

0    73
Name: 企业总评分, dtype: int64

In [132]:
def convert(df, error_per, diff_num):
    error_id = random.sample(range(1001, 4000), int(error_per*(3000)))
    for e_id in error_id:
        df.loc[(df['企业编号'] == e_id), '企业总评分'] = df.loc[(df['企业编号'] == e_id), '企业总评分']\
            + random.randint(-1*diff_num, diff_num)
    c, p = pearsonr(df['企业总评分'].values, real_data['企业总评分'].values)
    print('Random sample: %f' % error_per)
    print('Margin of error: %f' % diff_num)
    print('Mean squared error: %f' % mean_squared_error(df['企业总评分'].values, real_data['企业总评分'].values))
    print('Pearsonr: %f' % c)
    print('P: %f' % p)

In [143]:
real_data_change = real_data.copy()
convert(real_data_change, 0.8, 1)
series = real_data_change['企业总评分']
wb = load_workbook(filename=output_path)
ws = wb.worksheets[0]
for i in range(len(series)):
    ws.cell(row=i+1, column=2).value = int(series[i])
wb.save(output_path)
# pd.read_excel(output_path).head(n=20)
!java -jar ../../Evaluation/mission1_test.jar ../../Evaluation/mission1_real.xlsx ../../Evaluation/mission1_predict.xlsx

Random sample: 0.800000
Margin of error: 1.000000
Mean squared error: 0.543573
Pearsonr: 0.989674
P: 0.000000
0.0


|Random sample|Margin of error[-n,n]|Pearsonr|Mean squared error|jar Result|
|---|---|---|---|---|
|0.1 |1|0.99872|0.06678|0.00000|
|0.1 |3|0.99226 |0.40367 |0.00000|
|0.3 |2|0.98757 |0.65843 |0.00000|
|0.3 |3|0.97796 |1.19666 |1.00000|
|0.5 |1|0.99374 |0.32688 |0.00000|
|0.5 |3|0.96317 |2.02638 |1.41421|
|0.5 |5|0.91622 |5.07446 |2.23607|
|0.8 |1|0.98967 |0.54357 |0.00000|
|0.8 |2|0.97092 |1.56661 |1.00000|